Hi, sorry for the bugspam!  ;)  I don't mean for this to be my personal support forum but I suspect this may be an actual issue with some machines.
I have Pyqt4 and Pyqt5 installed and I believe this is messing up pyrpl.
I tried installing pyrpl from pip, and cloned the git respository, both with the same results: I get import errors for pyrpl:
That's from ipython. From regular python2:
I need PyQt4 for some older apps so I can't uninstall that... it seems pyrpl is getting confused because of this. Any ideas?
No prob, I am happy if we discuss all issues rather sooner than later.
its been a while i had these issues. if i remember correctly, this bug can be worked around by either loading pyqt4 or pyqt5 before attempting to load pyrpl. so maybe playing with these combinations will get you somewhere:
or something like that. At least i believe the error message should change.
Another thing to try in ipython before loading pyrpl:
good luck
Hi
Thanks, unfortunately none of those work for me. They all give the same error regarding API version, (I made sure to exit out of ipython each time I tried a new command set) except for the last one; apparently pyqtgraph requires qt4.
Exception: PyQtGraph requires either PyQt4 or PySide; neither package could be imported.
I figured it out. Since pyqtgraph requires qt4, that has to be used. That's fine.
So pyrpl/init.py line 28 should be (explicitly)
IPYTHON.magic("gui qt4")
In addition, to force the loading of qt4 and NOT qt5 on line 34, first do:
This forces qtpy to load qt4, otherwise it will by default load qt5.
Now purpyl imports fine.
(I accidentally closed this bug and don't know how to reopen! )
Cool thanks! only loading qt4 will not work since many people only have qt5. But maybe we can find another workaround, i will have to play around a little.
It's kind of weird, the above trick only works if I do the following in ipython BEFORE calling import pyrpl:
It apparently does no good changing the __init__ file to have the above lines, which makes no sense to me.
